lib/hawkular/operations/operations_api.rb

```diff
@@ -322,6 +339,8 @@ def update_collection_intervals(hash, &callback)
     private
 
     def invoke_operation_helper(operation_payload, operation_name = nil, binary_content = nil, &callback)
+      synchronize { connect }
```

Comment:
That's true, actually, and it is something that was not covered by tests, so I fixed it.